### ISSUE TITLE
`fill-mask` pipeline provides `<mask>` token among predictions

### ISSUE THREAD

**Issue by rspreafico-absci** [huggingface/transformers#12390]
## Environment info
- `transformers` version: 4.8.1
- Platform: Linux-5.4.0-74-generic-x86_64-with-glibc2.31
- Python version: 3.9.5
- PyTorch version (GPU?): 1.9.0+cu111 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: see below
- Using distributed or parallel set-up in script?: no

### Who can help
@LysandreJik
@sgugger

## Information

Model I am using (Bert, XLNet ...): RoBERTa

The problem arises when using:
* [X] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

```
from transformers import RobertaTokenizerFast, RobertaForMaskedLM, pipeline

tokenizer = RobertaTokenizerFast.from_pretrained("roberta-base")
model = RobertaForMaskedLM.from_pretrained("roberta-base")

fill_mask = pipeline(
    "fill-mask",
    model=model,
    tokenizer=tokenizer,
)

resp = fill_mask("My <mask> is Roberto.", top_k=len(tokenizer.get_vocab()))

[x for x in resp if x['token'] == tokenizer.mask_token_id]
```

## Expected behavior

Because the job of the `fill-mask` pipeline is to fill the `<mask>` special token, the expectation is that `<mask>` itself is not part of the possible predictions.
